source/defclass-star.lisp

```diff
@@ -522,3 +522,86 @@ See `define-class' for more details."
      `(define-condition ,name ,supers
         ,processed-slots
         ,@clean-options))))
+
+(defun generalize-arglist (arglist)
+  "Generalizes ARGLIST to be accepted by `defgeneric' (3.4.2 of CLCS/CLHS).
+Removes all the default forms or specializers and removes all keyword
+arguments if &allow-other-keys is present."
+  ;; FIXME: Is removing all the list-like args safe as generic
+  ;; function arglist?
```

Comment:
What are list-like args?

Reply:
Args like `((a specializer) (b specializer) &key (key value key-p))` etc.

Reply:
What's different?
In any case, maybe give an example?